internal/logger/logger_test.go

```diff
@@ -123,157 +117,201 @@ func validateLogOutputAtSpecifiedFormatAndSeverity(t *testing.T, format string,
 	}
 
 	// Assert warning and error logs are logged when log level is WARNING.
-	validateLogOutputAtSpecifiedFormatAndSeverity(t.T(), "text", cfg.WARNING, expected)
+	validateLogOutputAtSpecifiedFormatAndSeverity(t, "text", cfg.WARNING, expected)
 }
 
-func (t *LoggerTest) TestTextFormatLogs_LogLevelINFO() {
-	var expected = []string{
-		"", "", textInfoString, textWarningString, textErrorString,
+func TestTextFormatLogs_LogLevelINFO(t *testing.T) {
+	expected := []string{
+		"", // TRACE
+		"", // DEBUG
+		expectedLogRegex("text", "INFO", "www.infoExample.com"),
+		expectedLogRegex("text", "WARNING", "www.warningExample.com"),
+		expectedLogRegex("text", "ERROR", "www.errorExample.com"),
 	}
 
 	// Assert info, warning & error logs are logged when log level is INFO.
-	validateLogOutputAtSpecifiedFormatAndSeverity(t.T(), "text", cfg.INFO, expected)
+	validateLogOutputAtSpecifiedFormatAndSeverity(t, "text", cfg.INFO, expected)
 }
 
-func (t *LoggerTest) TestTextFormatLogs_LogLevelDEBUG() {
-	var expected = []string{
-		"", textDebugString, textInfoString, textWarningString, textErrorString,
+func TestTextFormatLogs_LogLevelDEBUG(t *testing.T) {
+	expected := []string{
+		"", // TRACE
+		expectedLogRegex("text", "DEBUG", "www.debugExample.com"),
+		expectedLogRegex("text", "INFO", "www.infoExample.com"),
+		expectedLogRegex("text", "WARNING", "www.warningExample.com"),
+		expectedLogRegex("text", "ERROR", "www.errorExample.com"),
 	}
 
 	// Assert debug, info, warning & error logs are logged when log level is DEBUG.
-	validateLogOutputAtSpecifiedFormatAndSeverity(t.T(), "text", cfg.DEBUG, expected)
+	validateLogOutputAtSpecifiedFormatAndSeverity(t, "text", cfg.DEBUG, expected)
 }
 
-func (t *LoggerTest) TestTextFormatLogs_LogLevelTRACE() {
-	var expected = []string{
-		textTraceString, textDebugString, textInfoString, textWarningString, textErrorString,
+func TestTextFormatLogs_LogLevelTRACE(t *testing.T) {
+	expected := []string{
+		expectedLogRegex("text", "TRACE", "www.traceExample.com"),
+		expectedLogRegex("text", "DEBUG", "www.debugExample.com"),
+		expectedLogRegex("text", "INFO", "www.infoExample.com"),
+		expectedLogRegex("text", "WARNING", "www.warningExample.com"),
+		expectedLogRegex("text", "ERROR", "www.errorExample.com"),
 	}
 
 	// Assert all logs are logged when log level is TRACE.
-	validateLogOutputAtSpecifiedFormatAndSeverity(t.T(), "text", cfg.TRACE, expected)
+	validateLogOutputAtSpecifiedFormatAndSeverity(t, "text", cfg.TRACE, expected)
 }
 
-func (t *LoggerTest) TestJSONFormatLogs_LogLevelOFF() {
-	var expected = []string{
-		"", "", "", "", "",
+func TestJSONFormatLogs_LogLevelOFF(t *testing.T) {
+	expected := []string{
+		"", // TRACE
+		"", // DEBUG
+		"", // INFO
+		"", // WARNING
+		"", // ERROR
 	}
 
 	// Assert that nothing is logged when log level is OFF.
-	validateLogOutputAtSpecifiedFormatAndSeverity(t.T(), "json", cfg.OFF, expected)
+	validateLogOutputAtSpecifiedFormatAndSeverity(t, "json", cfg.OFF, expected)
```

Comment:
These tests are not readable since the information about the Arrange, Act and Asserts blocks are all concealed in a function. I'd recommend we add them explicitly here even if that means duplicating the logic. [That's fine for tests](https://testing.googleblog.com/2019/12/testing-on-toilet-tests-too-dry-make.html).